### PR TITLE
load segment with target version, avoid read redundant segment

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
issue: #24171

### CODE REVIEW

**Comment by weiliu1031** [comment]
rerun ut

**Comment by weiliu1031** [comment]
rerun ut

File: internal/proto/query_coord.proto [hunk 1]
```diff
@@ -547,6 +548,7 @@ enum SyncType {
   Remove = 0;
   Set = 1;
   Amend = 2;
+  UpdateVersion = 3;
```
Comment (jiaoew1991) [inline internal/proto/query_coord.proto:551]: UpdateVersion is a complex operation, which is different from the atomic operations of remove, set and amend mentioned above.

File: internal/querynodev2/delegator/distribution.go [hunk 2]
```diff
@@ -176,6 +206,43 @@ func (d *distribution) AddOfflines(segmentIDs ...int64) {
...
+		entry, ok := d.sealedSegments[segmentID]
+		if !ok {
+			log.Error("readable sealed segment lost, make it unserviceable",
+				zap.Int64("segmentID", segmentID))
+			d.serviceable.Store(false)
+			continue
+		}
+		entry.TargetVersion = newVersion
+		d.sealedSegments[segmentID] = entry
+	}
```
Comment (congqixia) [inline internal/querynodev2/delegator/distribution.go:233]: These lines access the map without mutex protection, which may cause data race

File: internal/proto/query_coord.proto (hunk 1 above)
Comment (weiliu1031) [reply internal/proto/query_coord.proto:551]: UpdateVersion just like remove and set, will modify the segment distribution on shard delegator

**Pull request merged**